### DISCUSSION TITLE
Centralized logging / analytics / opt-in sharing production data and alert history

### DISCUSSION THREAD

**Discussion by wreiske** [jasonacox/Powerwall-Dashboard#222] [Ideas]
I have been experimenting with https://developers.cloudflare.com/analytics/analytics-engine/ to store time series data at a massive scale. I'm using it to track page load times and other metrics, inserting thousands of records every second and reporting on the data using Grafana https://developers.cloudflare.com/analytics/analytics-engine/grafana/

Today I woke up and thought, "Hmm. Wouldn't it be cool to (optionally / anonymously) push Powerall-Dashboard data to a central place, like cloudflare analytics engine, to track events across many Tesla solar systems?" 

The alerts/metrics are awesome... I'd be happy to share my production data and alerts and I'd be very curious to see what other systems are doing. Depending on how "anonymous" the data collection opt-in would be, some users could opt-in to share their postal code and see what alerts/production other users are doing in their region.

Sorry for the raw conscious dump here - I wanted to get the idea out before I moved on to the next thing and forgot about it!

tl;dr: Should we build an optional API endpoint for Powerwall-Dashboard users to send their metrics to so we can build some cool usage stats, alert stats, etc, across many systems?

**Comment by jasonacox** [comment]
Hi @wreiske - thanks for kicking off the discussion. I know that there are several of us who use the pvoutput tool to publish our data to the public [pvoutput.org](https://pvoutput.org/) site.  Here is more detail about that: https://github.com/jasonacox/Powerwall-Dashboard/discussions/46

The Cloudflare tools looks nice but I couldn't tell, is there a cost?
